superset-frontend/src/SqlLab/components/HighlightedSql.jsx

```diff
@@ -41,79 +35,86 @@ const propTypes = {
   shrink: PropTypes.bool,
 };
 
-class HighlightedSql extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      modalBody: null,
-    };
-  }
+function HighlightedSql({
+  sql,
+  rawSql,
+  maxWidth = 50,
+  maxLines = 5,
+  shrink = false,
+}) {
+  const [modalBody, setModalBody] = useState(null);
 
-  shrinkSql() {
-    const ssql = this.props.sql || '';
+  const shrinkSql = useCallback(() => {
+    const ssql = sql || '';
     let lines = ssql.split('\n');
-    if (lines.length >= this.props.maxLines) {
-      lines = lines.slice(0, this.props.maxLines);
+    if (lines.length >= maxLines) {
+      lines = lines.slice(0, maxLines);
       lines.push('{...}');
     }
     return lines
       .map(line => {
-        if (line.length > this.props.maxWidth) {
-          return `${line.slice(0, this.props.maxWidth)}{...}`;
+        if (line.length > maxWidth) {
+          return `${line.slice(0, maxWidth)}{...}`;
         }
         return line;
       })
       .join('\n');
-  }
+  }, [maxLines, maxWidth, sql]);
 
-  triggerNode() {
-    const shownSql = this.props.shrink
-      ? this.shrinkSql(this.props.sql)
-      : this.props.sql;
+  const triggerNode = useCallback(() => {
```

Comment:
I don't think these useCallback method wrappers are going to work in this case because we're executing this method in the same file, so memoizing it doesn't help.

Reply:
I was wondering something about this, I used a lot of these useCallbacks because the linter was suggesting them. Reading more about them, I wasn't sure that it was necessary, and there were some articles about how they can slow down service. You mentioned earlier that editing the linter made sense, would this also be a case for that?

Reply:
not sure about the linter. Let me know if you have a screenshot of it.

Reply:
Did some reading about this topic, and found: https://overreacted.io/a-complete-guide-to-useeffect/ to be pretty illuminating.